sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -271,7 +271,10 @@ private string GetAssetsJson()
             while (true)
             {
                 var assetsJsonPresent = File.Exists(Path.Combine(path, "assets.json"));
-                var isGitRoot = Directory.Exists(Path.Combine(path, ".git"));
+
+                // Check for root .git directory or, less commonly, a .git file for git worktrees.
+                string gitRootPath = Path.Combine(path, ".git");
+                var isGitRoot = Directory.Exists(gitRootPath) || File.Exists(gitRootPath);
```

Comment:
There shouldn't be. Worktrees - apart from this - are transparent to any git tools. I'd recommend making similar changes across the language repos. I've been using worktrees for years since they were introduced without issue except the occasional one like this where it's assumed ".git" is always a directory. It's also a file for git submodules, which we don't use.

Reply:
I don't think so. Given that the assets directory is emplaced with a directory, your path ascension logic should never be confused by it.